internal/controller/defaults.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"os"
+
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/utils/ptr"
+
+	csiv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+)
+
+var imageDefaults = map[string]string{
```

Comment:
@nb-ohad so you are expecting the shape of data be
```
data:
  provisioner: <image>
  [...]
  addons: <image>
```?

Where is the `images:` key? Can I drop that?

Not reviewing the rest atm.

Reply:
There is no "images" key? Why do you think we need the nesting? 
The point of the configmap was to map each image to a specific key

Reply:
ack, this was a misunderstanding that we needed an `images:` key.